src/ipc/StoreMap.cc

```diff
@@ -253,6 +253,18 @@ Ipc::StoreMap::peekAtReader(const sfileno fileno) const
     return NULL;
 }
 
+const Ipc::StoreMap::Anchor *
+Ipc::StoreMap::peekAtWriter(const sfileno fileno) const
+{
+    const Anchor &s = anchorAt(fileno);
+    if (s.writing())
+        return &s; // immediate access by lock holder so no locking
+    if (s.reading())
+        return nullptr;
+    assert(false); // must be locked for reading or writing
```

Comment:
Is this StoreMap class used exclusively by Transients table? if not, this assert is a bit risky - the non-transient caches using it may just have these pointing at regular cache entries with nobody using them at this time.

Same logic applies for peekAtReader()

Reply:
> Eduard: The assertion below does check this requirement.

... and that assertion requires an `s.reading()` check above it, answering the other question in Amos' comment:

> Amos: It is also not clear why peeking at the writer requires any info about a reader.

Reply:
Okay, so is there any reason other than this assert to why this method is tied to s.reading() ?

This makes more sense to me (assuming you really need the assert at all):

> if (s.writing()) return &s; // immediate access by lock holder so no locking
> assert(s.reading()); // must be locked for reading or writing
> return nullptr;

So the reader status is not relevant (always nullptr if no writer), you get the assert info (if needed), and optimized builds are not slowed down by checking read locks.

Reply:
Done.